domain/application/resource/fileresourcestore.go

```diff
@@ -37,20 +37,24 @@ func (f fileResourceStore) Put(
 	r io.Reader,
 	size int64,
 	fingerprint resource.Fingerprint,
-) (ResourceStorageUUID, error) {
+) (string, error) {
```

Comment:
**question:** Why removing the strong type here ? (and in its interface definition in resourcestore.go)

Reply:
Good point, have changed it back as discussed. 

Also as discuessed, there is a problem with the current resource store interface. With the current `Get`, `Put`, `Remove` interface, the caller has to worry about the link table between the store storage and the resource and they would need a switch statement to tell which link table to use. This breaks the abstraction of the store getter being he only place we have that switch.

Perhaps for a follow up PR we could consider an interface like this:
```go 
// ResourceStore provides a list of methods necessary for interacting with
// a store for the resource.
type ResourceStore interface {
	// Get returns an io.ReadCloser for a resource in the resource store.
	Get(
		ctx context.Context,
		path resourceStoragePath,
	) (r io.ReadCloser, size int64, err error)

	// Put stores data from io.Reader in the resource store at the
	// using the resourceUUID as the key.
	Put(
		ctx context.Context,
		path resourceStoragePath,
		r io.Reader,
		size int64,
		fingerprint resource.Fingerprint,
	) error

	// Remove removes an unlinked resource from storage. 
	// If the resource is still linked, an error is returned.
	Remove(
		ctx context.Context,
		path resourceStoragePath,
	) error

	// Link records that the stored blob is linked to a resource.
	Link(
		ctx context.Context,
		path resourceStoragePath,
		resourceUUID resource.UUID,
	) error
	
	// Unlink unlinks the stored blob from the resource UUID.
	Unlink(
		ctx context.Context,
		path resourceStoragePath,
		resourceUUID resource.UUID,
	) error
}
```

@manadart what are your opinions on this?

Reply:
I may not understand your intent here. 

The store should only be abstracting the logic of dealing with bytes. It seems to me this would mean service or state inside the store, which isn't how we want it to go - the service uses the store.

Also now that I look at state, it seems the intentions aren't quire right.
```go
// SetResource adds the resource to blob storage and updates the metadata.
func (st *State) SetResource(
	ctx context.Context,
	config resource.SetResourceArgs,
) (resource.Resource, error) {
	return resource.Resource{}, nil
}
```

The service should getting and using a store, so it will know the resource type, which will also inform how it stores metadata and linkage.

Reply:
The intention here was to abstract away the difference between container image resources and file resources, which I believe was the motivation behind the design of the store interface, though I'm not married to it.

> The store should only be abstracting the logic of dealing with bytes. It seems to me this would mean service or state inside the store, which isn't how we want it to go - the service uses the store.

This is not what we have currently, the `Put` method of store returns the foreign key for the resource store link table to link against.

If instead the linkage is to be handled by the state layer and a switch statement put on the type then the current design suffices and there is not need to add those methods. 

It looks like the comment on `SetResource` is wrong, I believe these methods originally were copied directly from the service so some of the comments are still the same.